core/src/main/scala/zio/temporal/workflow/ZWorkflowQueue.scala

```diff
@@ -0,0 +1,61 @@
+package zio.temporal.workflow
+
+import io.temporal.workflow.{QueueConsumer, WorkflowQueue}
+import zio.Duration
+
+final class ZWorkflowQueue[E](val toJava: WorkflowQueue[E]) extends ZQueueConsumer[E](toJava) with ZQueueProducer[E] {
```

Comment:
Is it fine for you to keep all these 3 wrappers in the same file? I'll be happy to move them into separate files if you want.

Reply:
Now, looking at the wrapper, I wonder if adding it is worth it. Do you see any possible Scala-specific extensions we can provide? I mean additional methods, supporting Scala-specific types (besides `zio.Duration`), etc. 

Currently, this wrapper brings proper (in)variance checks and a `.map` method overload for Scala lambdas. What else can we do? Depending on the answer, we'd decide on what to do with those interfaces/wrappers.

Reply:
That's a good question. It is also worth adding it because of additional null checks for some methods that return null. Also, some methods throw exceptions, and that's the question I was going to ask you. Is allowing methods to raise exceptions in a workflow code semantically correct? I mean, we can additionally wrap the results of some methods in `Try`.

Reply:
Unfortunately, the workflow code should be built without `Try`, `Either`, or any other container. Temporal's error-handling logic can only work with Exceptions. If we wrap an error into `Try`, Temporal won't retry the workflow unless you `get`. Therefore, the existing `zio-temporal` functionality related to the Workflow code (such as activity invocations) doesn't wrap error-prune code but raises an exception

Reply:
Okay, got it. So, deciding whether we should have a wrapper is up to you. I don't see any other benefits besides what we already mentioned.

Reply:
It sounds enough to me. We can add more methods later 😃